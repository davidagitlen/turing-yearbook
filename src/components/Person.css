.person {
  display: flex;
  flex-direction: column;
  height: 475px;
  width: 88%;
  background-color: rgba(225, 225, 225, .85);
  border-radius: 25px;
  justify-content: center;
  align-items: center;
}

img {
  height: 60%;
  width: 87%;
  border-radius: 25px;
}

p {
  font-style: italic;
}

span {
  font-weight: bold;
  color: blue;
}