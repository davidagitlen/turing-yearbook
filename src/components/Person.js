import React, {Component} from 'react';
import './Person.css';

class Person extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: this.props.name,
      quote: this.props.quote,
      superlative: this.props.superlative,
    };
  }

  handleChange = (event) => {
    this.setState({[event.target.getAttribute('name')]: event.target.innerText });
  }

  render() {
    return (
      <div className='person' data-id={this.props.id}>
        <img src={this.props.photo} alt={this.props.name + (' looking magnificent')}/>
        <h3 
          contentEditable={this.props.cohort === 'student' ? 'true' : 'false'}
          suppressContentEditableWarning='true' 
          name='name'
          onKeyUp={event => this.handleChange(event)}
          >{this.props.name}</h3>
        <p 
          contentEditable={this.props.cohort === 'student' ? 'true' : 'false'}
          suppressContentEditableWarning='true' 
          name='quote'
          onKeyUp={event => this.handleChange(event)}
          >{this.props.quote}</p>
        <span 
          contentEditable={this.props.cohort === 'student' ? 'true' : 'false'}
          suppressContentEditableWarning='true' 
          name='superlative'
          onKeyUp={event => this.handleChange(event)}
          >{this.props.superlative}</span>
        <button 
          onClick={() => this.props.deleteStudent(this.props.id)}
          className={this.props.cohort === "student" ? "" : "display-none"}
          >Delete Me!</button>
      </div>
    )
  }
}

export default Person;
