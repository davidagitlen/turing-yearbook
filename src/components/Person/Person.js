import React, {Component} from 'react';
import './Person.css';

class Person extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: this.props.id,
      name: this.props.name,
      quote: this.props.quote,
      superlative: this.props.superlative,
    };
  }

  handleChange = (event) => {
    this.setState({[event.target.getAttribute('name')]: event.target.innerText });
  }

  render() {
    const editableClass = this.props.cohort === 'student' ? 'true' : 'false';
    const displayClass = this.props.cohort === "student" ? "" : "display-none"; 

    return (
      <div className='person' data-id={this.props.id}>
        <img src={this.props.photo} alt={this.props.name + (' looking magnificent')}/>
        <h3 
          contentEditable={editableClass}
          suppressContentEditableWarning='true' 
          name='name'
          onKeyUp={event => this.handleChange(event)}
          onBlur={() => this.props.editStudent(this.state)}
          >{this.props.name}</h3>
        <p 
          contentEditable={editableClass}
          suppressContentEditableWarning='true' 
          name='quote'
          onKeyUp={event => this.handleChange(event)}
          onBlur={() => this.props.editStudent(this.state)}
          >{this.props.quote}</p>
        <span 
          contentEditable={editableClass}
          suppressContentEditableWarning='true' 
          name='superlative'
          onKeyUp={event => this.handleChange(event)}
          onBlur={() => this.props.editStudent(this.state)}
          >{this.props.superlative}</span>
        <button 
          onClick={() => this.props.deleteStudent(this.props.id)}
          className={displayClass}
          >Delete Me!</button>
      </div>
    )
  }
}

export default Person;
