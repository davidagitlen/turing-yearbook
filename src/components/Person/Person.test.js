import React from 'react';
import { shallow } from 'enzyme';
import people from '../../data/yearbook-data.js';
import Person from './Person';

let testPerson, wrapper; 

describe('Person', () => {

  it('should call editStudent prop with the Card\'s edited state when an edited element loses focus', () => {
    const editStudentMock = jest.fn();
    const testPerson = people.students[0];
    const wrapper = shallow(<Person 
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
      editStudent={editStudentMock}
    />);
    expect(wrapper.state().name).toEqual('Chris');
    wrapper.setState({name : 'Change!'});
    wrapper.find('h3').simulate('blur');
    expect(editStudentMock).toHaveBeenCalledWith({
      id : 21,
      name: 'Change!',
      quote: 'I love kicking puppies. Also, if I were a pokemon I would be clefairy!',
      superlative: 'Most Likely to Kick a Puppy'
    });
    expect(wrapper.state().name).toEqual('Change!');
  });

  it('should fire handleChange on keyup when an element has been edited', () => {
    const eventMock = { 
      target: { 
        name: 'name', 
        getAttribute: function () {return this['name']},
        innerText: 'test' }
      };
    const testPerson = people.students[0];
    const wrapper = shallow(<Person
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
    />);
    wrapper.instance().handleChange = jest.fn();
    wrapper.update();
    wrapper.find('h3').simulate('keyup', eventMock);
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });

  it('should call deleteStudent prop with the person\'s id when the button is clicked', () => {
    const deleteStudentMock = jest.fn();
    const testPerson = people.students[0];
    const wrapper = shallow(<Person
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
      deleteStudent={deleteStudentMock}
    />);
    wrapper.find('button').simulate('click');
    expect(deleteStudentMock).toHaveBeenCalledWith(21);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    const testPerson = people.students[0];
    const wrapper = shallow(<Person
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if the card\'s cohort prop is student', () => {
    const testPerson = people.students[0];
    const wrapper = shallow(<Person
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
      cohort='student'
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if the card\'s cohort prop is staff', () => {
    const testPerson = people.staff[0];
    const wrapper = shallow(<Person
      photo={testPerson.photo}
      name={testPerson.name}
      quote={testPerson.quote}
      superlative={testPerson.superlative}
      id={testPerson.id}
      cohort='staff'
    />);

    expect(wrapper).toMatchSnapshot();
  });
  
})