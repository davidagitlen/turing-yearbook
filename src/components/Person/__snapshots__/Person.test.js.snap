// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot if the card's cohort prop is staff 1`] = `ShallowWrapper {}`;

exports[`Person should match the snapshot if the card's cohort prop is student 1`] = `ShallowWrapper {}`;

exports[`Person should match the snapshot with all data passed in correctly 1`] = `ShallowWrapper {}`;
